lib/WooCommerce/Subscription.php

```diff
@@ -111,6 +116,29 @@ public function subscribeOnCheckout($orderId, $data) {
     return true;
   }
 
+  private function updateAlreadySubscribedSubscriber(Subscriber $subscriber) {
+    $subscriber->status = Subscriber::STATUS_SUBSCRIBED;
+    if (empty($subscriber->confirmedIp) && empty($subscriber->confirmedAt)) {
+      $subscriber->confirmedIp = Helpers::getIP();
+      $subscriber->setExpr('confirmed_at', 'NOW()');
+    }
+    $subscriber->save();
+  }
+
+  private function updateNotSubscribedSubscriber(Subscriber $subscriber) {
+    $signupConfirmation = $this->settings->get('signup_confirmation');
+    if ((bool)$signupConfirmation['enabled'] === false) {
+      // if double opt-in is disabled we act as if the subscriber was already subscribed
+      $this->updateAlreadySubscribedSubscriber($subscriber);
+      return;
+    }
```

Comment:
It would make more sense to do this test before calling this method. Doing it on the `subscribeOnCheckout` method and calling `updateAlreadySubscribedSubscriber` directly from there seems more reasonable.